main.nf

```diff
@@ -214,35 +201,35 @@ process bwa {
 
     module 'bioinfo-tools'
     module 'bwa'
-    module 'bamtools'
     module 'samtools'
     module 'BEDTools'
-
+    
     cpus 2
-    memory '16 GB'
-    time '120h'
-
-    publishDir "$results_path/bwa"
-
+    memory { 16.GB * task.attempt }
+    time { 120.h * task.attempt }
+    errorStrategy { task.exitStatus == 143 ? 'retry' : 'terminate' }
+    maxRetries 3
+    maxErrors '-1'
+    
+    publishDir "${params.outdir}/bwa", mode: 'copy'
+    
     input:
     file (reads:'*') from trimmed_reads
     set val(prefix) from name_for_bwa
 
     output:
-    file '*.sort.bam' into bam_picard
-    file '*.sort.bam.bai' into results
-    file '*.sort.bed' into results
-
+    file '*.sorted.bam' into bam_picard
+    file '*.sorted.bam.bai'
+    file '*.sorted.bed'
+    stdout into bwa_logs
+    
+    script:
     """
-    bwa mem -M $genomebwa ${reads} > ${prefix}.sam
-    samtools view -bT $genomebwa ${prefix}.sam > ${prefix}.bam
-    samtools sort ${prefix}.bam ${prefix}.sort
-    samtools index ${prefix}.sort.bam
-    rm ${prefix}.sam
-    rm ${prefix}.bam
-    bedtools bamtobed -i ${prefix}.sort.bam | sort -k 1,1 -k 2,2n -k 3,3n -k 6,6 > ${prefix}.sort.bed
+    bwa mem -M $index $reads | samtools view -b > ${prefix}.bam
+    samtools sort ${prefix}.bam ${prefix}.sorted
```

Comment:
Also, I think I would prefer to separate the alignment from the sort/index/bamtobed, so that we have two processes. They're sort of distinct in what they're doing, plus the alignment is very slow. If the sort/index fails then we lose the alignments currently. Having them in two processes should be more robust.